enhancements/builds/volume-mounted-resources.md

```diff
@@ -0,0 +1,374 @@
+
+- Node pull secrets (host)
+- System configuration (ConfigMap)
+- Certificate authorities for the cluster proxy and internal registry (ConfigMap)
+- Blob metadata cache (host)
+- RHEL entitlements (host via cri-o configuration)
+
+In addition, the builder process generates the following buildah volume mounts:
+
+- RHEL entitlements (destination is `/run/secrets/etc-pki-entitlement`, `/run/secrets/redhat.repo`, and `/run/secrets/rhsm`)
+- Custom PKI trust bundle - destination is `/etc/pki/ca-trust`, added via the Build/BuildConfig's `mountTrustedCA` option.
+
+If an input volume's name collides with a volume created by the build controller, or implied by OpenShift's cri-o configuration, the build should fail to run.
+If a volume mount path collides with a buildah volume mount generated by the builder process, the build should fail to run.
+
+#### Failure behavior
+
+A BuildConfig object can reference a Secret or ConfigMap in a volume that does not exist in the BuildConfig's namespace (yet).
+If a Build is generated from a BuildConfig and a referenced Secret or ConfigMap does not exist, the build controller should report a failure status and not create a build pod.
+This behavior should also apply to Secrets or ConfigMaps referenced in the build source array.
+
+### Risks and Mitigations
+
+**Risk:** Build mounts can alter the behavior of the build itself (a security risk).
+
+*Mitigations:*
+
+- Supported volume types are gated by the API.
+  For example, `HostPath` volume mounts are not supported.
+- Input volumes are mounted in a fixed location within the build pod's container.
+  The container volume is created by the build controller and cannot be changed by an end user.
+- Builds fail to run if a volume name collides with a volume generated by the build controller.
+- Builds fail if the a volume mount path collides with a destination generated by the builder process.
+
+**Risk:** Supporting arbitrary volume sources can lead to privilege escalations.
+This is a particular concern for builds since build pods run privileged.
+
+*Mitigations:*
+
+- Only volume sources that are known to provide storage isolated from the host file system will be supported (Secrets and ConfigMaps).
+- Future supported volume types must ensure isolation from the host.
+  Gating must happen at the API or build controller config level, since the build controller uses the privileged SCC to create pods.
+
+## Design Details
+
+## Open Questions [optional]
+
+1. Can we make this the default or even only behavior for builds?
+   No, need to make it opt-in to avoid potentially breaking existing buildconfigs.
+2. What happens if a user overrides the default volume mounts used by Builds today?
+   We prevent the build from running.
+   Overriding default volume mounts can alter build behavior, which can be a security risk.
+3. Can cluster admins alter the default `mounts.conf` configuration for cri-o?
```

Comment:
/cc @mrunalp 

I believe we use `mounts.conf` to get RHEL entitlements into workloads from the host, which is relevant for a) RHEL7 nodes, and b) clusters that use MacineConfig to add RHEL entitlements to nodes.

In both situations, these are work-arounds for our lack of entitlements support in RHCOS.

Reply:
@adambkaplan Today we don't allow cluster admins to modify the mounts.conf through ContaineRuntineConfig CRD. They can of course directly modify any such file using the MCO but that is unsupported.